src/Electron/electron.ts

```diff
@@ -188,6 +188,22 @@ export class ElectronContainerWindow extends ContainerWindow {
         });
     }
 
+    public getState(): Promise<any> {
+        if (this.innerWindow && this.innerWindow.webContents) {
+            return this.innerWindow.webContents.executeJavaScript("getState ? getState() : undefined");
+        } else {
+            return Promise.resolve(undefined);
+        }
+    }
+
+    public setState(state: any): Promise<void> {
+        if (this.innerWindow && this.innerWindow.webContents) {
+            return this.innerWindow.webContents.executeJavaScript(`if (setState) setState(JSON.parse(\`${JSON.stringify(state)}\`))`);
```

Comment:
We sure there is no better expression of the intent here?

Reply:
@psmulovics  What do you mean by intent?

Reply:
It's certainly not pretty but there is no access to the underlying native window (cross process) and doing it via ipc would require having two way send but if the otherwise doesn't have any code in place to handle it the caller would be waiting for a reply that isn't coming.

Reply:
`if (setState)` would throw an error if `setState` isn't defined; should this be `if (typeof setState !== "undefined")` or `if (window.setState)` ?

Reply:
@1j01 I assumed since it was in the global that window.setState is equivalent to setState.  Testing it is a bit muddied since it goes through executeJavaScript which can be masking this from me.  I'll update it if not for a fix but definitely for clarity.

Reply:
@1j01 @psmulovics Updated to add explicit use of methods on window global and formatting with braces where appropriate.